### PR TITLE
fix(Variance): Add better prop sorting tests

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Found some borked behavior with the previous tests just being wrong

Added a better safeguard here for shorthand properties and a more exhaustive test to ensure that the most general shorthands are ordered first to ensure the cascade behaves as expected border is above borderColor to prevent incidental overrides.

This ensures that within the same scope worrying about the cascade is not important.  This is necessary to determine due to composition being ad hoc, having predictability when users may not have full access to cascading beyond merge order.

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.10-alpha.748ee8.0<br/>@codecademy/gamut-kit@0.3.8-alpha.748ee8.0<br/>@codecademy/gamut-labs@13.3.8-alpha.748ee8.0<br/>@codecademy/gamut-styles@9.3.1-alpha.748ee8.0<br/>@codecademy/gamut-tests@2.3.7-alpha.748ee8.0<br/>@codecademy/gamut@29.9.3-alpha.748ee8.0<br/>@codecademy/konamimojisplosion@0.2.7-alpha.748ee8.0<br/>@codecademy/markdown-overrides@0.5.20-alpha.748ee8.0<br/>@codecademy/styleguide@33.8.5-alpha.748ee8.0<br/>@codecademy/variance@0.7.1-alpha.748ee8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070b925b7ad0d4bc670e167--gamut-preview.netlify.app>https://6070b925b7ad0d4bc670e167--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070b925b7ad0d4bc670e167>Deploy Logs</a></p>

File: packages/variance/src/utils/__tests__/propNames-test.ts [hunk 1]
```diff
@@ -1,5 +1,104 @@
 import { orderPropNames } from '../propNames';
 
+const test = {
+  textColor: { property: 'color' },
```
Comment (JoshuaKGoldberg) [inline packages/variance/src/utils/__tests__/propNames-test.ts:4]: I wonder if this big thing is going to end up being reused, and therefore would belong in a fixtures directory... probably not worth it in this PR.

**Review by JoshuaKGoldberg — APPROVED** [review]
Works for me :)

File: packages/variance/src/utils/__tests__/propNames-test.ts (hunk 1 above)
Comment (codecaaron) [reply packages/variance/src/utils/__tests__/propNames-test.ts:4]: Good call :)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.10-alpha.262401.0<br/>@codecademy/gamut-kit@0.3.8-alpha.262401.0<br/>@codecademy/gamut-labs@13.3.8-alpha.262401.0<br/>@codecademy/gamut-styles@9.3.1-alpha.262401.0<br/>@codecademy/gamut-tests@2.3.7-alpha.262401.0<br/>@codecademy/gamut@29.9.3-alpha.262401.0<br/>@codecademy/konamimojisplosion@0.2.7-alpha.262401.0<br/>@codecademy/markdown-overrides@0.5.20-alpha.262401.0<br/>@codecademy/styleguide@33.8.5-alpha.262401.0<br/>@codecademy/variance@0.7.1-alpha.262401.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070c2d1fa22a45f1c4a9fc6--gamut-preview.netlify.app>https://6070c2d1fa22a45f1c4a9fc6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070c2d1fa22a45f1c4a9fc6>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.10-alpha.1b5e69.0<br/>@codecademy/gamut-kit@0.3.8-alpha.1b5e69.0<br/>@codecademy/gamut-labs@13.3.8-alpha.1b5e69.0<br/>@codecademy/gamut-styles@9.3.1-alpha.1b5e69.0<br/>@codecademy/gamut-tests@2.3.7-alpha.1b5e69.0<br/>@codecademy/gamut@29.9.3-alpha.1b5e69.0<br/>@codecademy/konamimojisplosion@0.2.7-alpha.1b5e69.0<br/>@codecademy/markdown-overrides@0.5.20-alpha.1b5e69.0<br/>@codecademy/styleguide@33.8.5-alpha.1b5e69.0<br/>@codecademy/variance@0.7.1-alpha.1b5e69.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070ca40144bb400a470885a--gamut-preview.netlify.app>https://6070ca40144bb400a470885a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070ca40144bb400a470885a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.10-alpha.d797d7.0<br/>@codecademy/gamut-kit@0.3.9-alpha.d797d7.0<br/>@codecademy/gamut-labs@13.4.1-alpha.d797d7.0<br/>@codecademy/gamut-styles@9.3.1-alpha.d797d7.0<br/>@codecademy/gamut-tests@2.3.7-alpha.d797d7.0<br/>@codecademy/gamut@29.9.3-alpha.d797d7.0<br/>@codecademy/konamimojisplosion@0.2.7-alpha.d797d7.0<br/>@codecademy/markdown-overrides@0.5.20-alpha.d797d7.0<br/>@codecademy/styleguide@33.9.1-alpha.d797d7.0<br/>@codecademy/variance@0.7.1-alpha.d797d7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6070cc35d2f5ec00a52af474--gamut-preview.netlify.app>https://6070cc35d2f5ec00a52af474--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6070cc35d2f5ec00a52af474>Deploy Logs</a></p>

**Pull request merged**